app/services/course/assessment/question/programming_codaveri/java/java_package_service.rb

```diff
@@ -0,0 +1,141 @@
+      test_case_object[:timeout] = @question.time_limit * 1000 if @question.time_limit
+      test_case_object[:prefix] = cleaned_prefix
+      test_case_object[:lhsExpression] = expression
+      test_case_object[:rhsExpression] = expected
+      test_case_object[:display] = expression
+
+      @test_case_files.append(test_case_object)
+    end
+  end
+
+  def extract_supporting_files
+    extract_supporting_main_files
+    extract_supporting_tests_files
+  end
+
+  def extract_supporting_main_files
+    main_files = @package.main_files.compact.to_h
+    main_filenames = main_files.keys
+
+    main_filenames.each do |filename|
+      next if ['Makefile', 'build.xml', '.meta'].include?(filename.to_s)
+
+      extract_supporting_file(filename, main_files[filename])
+    end
+  end
+
+  def extract_supporting_tests_files
+    test_files = @package.test_files
+    test_filenames = test_files.keys
+
+    test_filenames.each do |filename|
+      next if ['append', 'prepend', 'autograde', 'RunTests.java'].include?(filename.to_s)
+
+      extract_supporting_file(filename, test_files[filename])
+    end
+  end
+
+  def extract_supporting_file(filename, content)
+    supporting_solution_object = default_codaveri_data_file_template
+
+    supporting_solution_object[:type] = 'internal'
+    supporting_solution_object[:path] = filename.to_s
+    supporting_solution_object[:content] = content
+
+    @data_files.append(supporting_solution_object)
+  end
+
+  def extract_template
+    submission_files = @package.submission_files
+    test_files = @package.test_files
+
+    submission_files.each_key do |pathname|
+      main_template_object = default_codaveri_template_template
+
+      main_template_object[:path] = extract_pathname_from_file(submission_files, pathname)
+      main_template_object[:content] = submission_files[pathname]
+      main_template_object[:prefix] = strip_autograding_definition_from(test_files[Pathname.new('prepend')])
+      # TODO: fill in the suffix properly when we have aligned our append file convention with Codaveri
+      main_template_object[:suffix] = ''
+
+      @template_files.append(main_template_object)
+    end
+  end
+
+  def preload_question_test_cases
+    # The regex below finds all text after the last slash
+    # (eg AutoGrader/AutoGrader/test_private_4 -> test_private_4)
+    @question.test_cases.pluck(:identifier, :id).to_h { |x| [x[0].match(/[^\/]+$/).to_s, x[1]] }
+  end
+
+  def extract_pathname_from_file(submission_files, pathname)
+    class_name_extractor = /public\s+class\s+([A-Za-z_][A-Za-z0-9_]*)/
+    match = submission_files[pathname].match(class_name_extractor)
+
+    match ? "#{match[1]}.java" : pathname.to_s
+  end
+
+  def strip_autograding_definition_from(file_content)
+    # we strip away all the definitions inside the Autograder class defined within prepend,
+    # which has 6256 characters. Those definitions are not needed to be sent to Codaveri
```

Comment:
nit: you should also indicate the file you are referencing

Reply:
done